### ISSUE TITLE
can I apply this tutorial to Blazor WebAssembly?  

### ISSUE THREAD

**Issue by gostan99** [makma/statiq-tutorial#4]

**Comment by makma** [comment]
Hi @gostan99, please find the answer provided by @daveaglick (creator of Statiq) bellow.

In theory, a Blazor app using the [WebAssembly hosting model](https://docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-5.0#blazor-webassembly) should be able to co-exist with a Statiq site (or really any static site, including output from Hugo, Jekyll, etc. That's because the Blazor WebAssembly hosting model is itself just a bunch of static resources that get delivered to the client. The challenge is that going from Blazor source code to the final compiled static Blazorr WebAssembly resources requires a fair amount of black-box tooling baked into .NET. I.e. a compilation phase has to happen somewhere to get from your Blazor code to something that can live alongside a static site.
​
Right now that would be on the developer to figure out - there's no built-in Blazor compilation support in Statiq right now. It might be possible to compile the Blazor app first, copy its output to the Statiq input folder, and then run Statiq (which would output the Blazor files just like any other static resources). It might also be possible to kick off a Blazor compilation alongside the Statiq generation from the command line as a single unit of work. I've also got it in the back of my head that at some point it would be cool to look at an integration (along the lines of how Razor works in Statiq) - that's probably a ways off though.
​
TL;DR: there's nothing in the box right now for combining the two, and while it's probably possible, it wouldn't be straightforward today.
